### DISCUSSION TITLE
compile failure

### DISCUSSION THREAD

**Discussion by yonglbl** [GMOD/jbrowse-components#4296] [Q&A] [answered]
hello, when i run my react app useing the react-linear-genome-view component, it shows errors when compile, any suggestion if I missing something?
ERROR in ./node_modules/@gmod/cram/esm/io/index.js 1:0-22
Module not found: Error: Can't resolve 'url' in '/srv/app/node_modules/@gmod/cram/esm/io'

BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
This is no longer the case. Verify if you need this module and configure a polyfill for it.

**Comment by cmdcolin** [comment, marked as answer]
you need a couple polyfills to properly use @jbrowse/react-linear-genome-view and other embedded components in webpack/create-react-app v5

we have some examples of setups here https://jbrowse.org/jb2/docs/embedded_components/

this source code uses create-react-app v5 with CRACO to supply the polyfills via the NodePolyfillPlugin https://github.com/GMOD/jbrowse-react-linear-genome-view-cra5-demo

an ongoing project is to reduce the need for polyfills in our app but currently, it is required :)

**Reply by yonglbl** [reply]
hello, thanks for the answer, 
then, not sure is related to installation, when I run it, based on the code from jbrowse-react-linear-genome-view-cra5-demo, 
there is error says:
ERROR
Cannot destructure property 'session' of 'viewState' as it is undefined.
TypeError: Cannot destructure property 'session' of 'viewState' as it is undefined.
    at http://localhost:8080/static/js/bundle.js:58475:5

JBrowseLinearGenomeView.js:9 Uncaught TypeError: Cannot destructure property 'session' of 'viewState' as it is undefined.
    at JBrowseLinearGenomeView.js:9:1
    
const JBrowseLinearGenomeView = observer(function ({ viewState, }) {
    const { session } = viewState;

**Reply by cmdcolin** [reply]
@yonglbl take note of this if statement that blocks rendering until viewState becomes defined https://github.com/GMOD/jbrowse-react-linear-genome-view-cra5-demo/blob/main/src/App.tsx#L38-L40
